cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -536,6 +536,37 @@ private void writeToHeartBeatTable(@Nullable RunRecordDetail recordedRunRecord,
         // Publish the program STARTING state before starting the program
         programStateWriter.start(programRunId, newProgramOptions, null, programDescriptor);
 
+        long provisioningTime = System.currentTimeMillis() / 1000 -
+          RunIds.getTime(programRunId.getRun(), TimeUnit.SECONDS);
+        // emit provisioning time metric
+        ProfileId profileId = null;
+        try {
+          RunRecordDetail runRecordMeta = appMetadataStore.getRun(programRunId);
```

Comment:
The `recordProgramProvisioned` method call already returns the latest `RunRecordMeta` being recorded. You shouldn't need to read it again.

Reply:
removed, as it's not needed.